### DISCUSSION TITLE
Configuring pyright LSP

### DISCUSSION THREAD

**Discussion by p5a0u9l** [LazyVim/LazyVim#1416] [Q&A] [answered]
Hi, I've read through the docs, looked at other's for examples, searched this forum, and am struggling to find an answer that works. 

I'm using python/pyright. The `.extras.lang.python` didn't seem to have the right configuration for my use. So I've set it up, per the docs, under `plugins/lsp.lua`. 

I'm trying to disable pyright's type checking - `typeCheckingMode = "off"`. 

```lua
      opts = {
        servers = {
          jsonls = {},

          -- pyright will be automatically installed with mason and loaded with lspconfig
          pyright = {
            settings = {
              python = {
                analysis = {
                  autoSearchPaths = true,
                  diagnosticMode = "openFilesOnly",
                  useLibraryCodeForTypes = true,
                  typeCheckingMode = "off",
                },
              },
            },
          },

          ruff_lsp = {},
        },
```
which doesn't seem to be doing anything. pyright works, but as far as I can tell I'm getting defaults only.

**Comment by dpetka2001** [comment, marked as answer]
If [these](https://github.com/p5a0u9l/dotfiles/blob/15fa168f161278ab694a8ffdbc890c4ab0c6075b/dots/nvim/lua/plugins/enabled.lua#L59-78) are your dotfiles, then you define `opts` inside `opts`. No need to do that. Inside the first `opts` just define `servers`. Right after `capabilities`.

**Reply by p5a0u9l** [reply]
Oh gosh, what a noob :) thank you, this fixed it!
